### PR TITLE
fix(#1990): changes priority of target element passed to popper component

### PR DESCRIPTION
**Pull request by watinha**
- [x] Bug fix 
- [ ] New feature 
- [ ] Chore 
- [ ] Breaking change 
- [x] There is an open issue which this change addresses
- [x] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** document.
- [x] My commits follow the [Git Commit Guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] My change requires a change to [Typescript typings](./types/lib).
  - [ ] I have updated the typings accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/TooltipPopoverWrapper.js [hunk 1]
```diff
@@ -351,7 +351,7 @@ class TooltipPopoverWrapper extends React.Component {
     return (
       <PopperContent
         className={className}
-        target={this.currentTargetElement || this._targets[0]}
+        target={this._targets[0] || this.currentTargetElement}
```
Comment (kyletsang) [inline src/TooltipPopoverWrapper.js:354]: Unfortunately, this would reintroduce #1185.

Comment (watinha) [reply src/TooltipPopoverWrapper.js:354]: @kyletsang, I am thinking I could try to use the focused element to identify which of the _targets is associated to it. Do you think this approach might solve this issue?

Comment (kyletsang) [reply src/TooltipPopoverWrapper.js:354]: The issue here is with the button click.  `TooltipPopoverWrapper` adds the click listener to `document`, so the code below

https://github.com/reactstrap/reactstrap/blob/a97a8345f3998e9388d42550b8e70aaa62a5be50/src/TooltipPopoverWrapper.js#L166

would result in `this.currentTargetElement` being set to `e.target` due to `e.currentTarget` being null.

I don't believe we should ever be using `e.target` here, so we should modify the code to scan the current list of targets and determine which target `e.target` falls under. This should only be done if `e.currentTarget` is null.

Comment (watinha) [reply src/TooltipPopoverWrapper.js]: Hey @kyletsang, I tried implementing exactly what you said. Please, let me know if this was the implementation you were expecting.

Comment (kyletsang) [reply src/TooltipPopoverWrapper.js]: Looks good!  Thanks

**Review by kyletsang — APPROVED** [review]

**Pull request merged**